### PR TITLE
TASK1

### PR DESCRIPTION
**Pull request by LykashevichAndrei**

### CODE REVIEW

File: task_DEV-1/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (IgorKheidorov) [inline task_DEV-1/Properties/launchSettings.json]: Whats the file? why do you push it on github? I think it is not necessary

File: task_DEV-1/Program.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    /// searches maximum number of same serial symbols in the string
+    /// and searches some exceptions
+    /// </summary>
+    class MaxNumberOfCharactersInString
+    {
+        public static void Main(string[] arg)
+        {
+            try
+            {
+                MaxNumberOfCharactersInString example = new MaxNumberOfCharactersInString(); //create instance of a class
```
Comment (IgorKheidorov) [inline task_DEV-1/Program.cs]: example is not a perfect name for MaxNumberOfCharactersInString class instance

File: task_DEV-1/Program.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        }
+        private int FindMaxSequnce(char[] str) // this function search the longest sequence
+        {
+            int max_length = 0;
+            int buf = 0;
+            for (int i = 0; i < str.Length - 1; i++)
+            {
+                if (str[i] == str[i + 1])
+                    buf++;//increase the values of the buffer ,if the symbols match
```
Comment (IgorKheidorov) [inline task_DEV-1/Program.cs]: code style violation, use {}

File: task_DEV-1/Program.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            }
+            catch
+            {
+                Console.WriteLine("ERROR");
+                Console.WriteLine("the function takes too many arguments");
+            }
+        }
+        private int FindMaxSequnce(char[] str) // this function search the longest sequence
+        {
+            int max_length = 0;
```
Comment (IgorKheidorov) [inline task_DEV-1/Program.cs]: Bad name, what is buf

File: task_DEV-1/Program.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            {
+                if (str[i] == str[i + 1])
+                    buf++;//increase the values of the buffer ,if the symbols match
+                else
+                    buf = 0;//equate to zero if the symbols don't match
+                if (max_length < buf)
+                {
+                    max_length = buf;
+                }
+            }
```
Comment (IgorKheidorov) [inline task_DEV-1/Program.cs]: Why +1? unclear

File: task_DEV-1/Program.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            return (max_length + 1);
+        }
+        private void Result(int length) //output result's
+        {
+            if (length == 1)
```
Comment (IgorKheidorov) [inline task_DEV-1/Program.cs]: Code style

File: task_DEV-1/Program.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+
+namespace Task1
+{   /// <summary>
+    /// Class MaxNumberOfCharactersInString
+    /// searches maximum number of same serial symbols in the string
+    /// and searches some exceptions
+    /// </summary>
+    class MaxNumberOfCharactersInString
```
Comment (IgorKheidorov) [inline task_DEV-1/Program.cs]: one class one file, code style violation

File: task_DEV-1/FindMaxSequnceInString.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
+﻿using Task1;
+
+namespace equal_sumbols
+{
+    class FindMaxSequnceInString
+    {
+        private int FindMaxSequnce(char[] str) // this function search the longest sequence
```
Comment (AlexanderSoroka) [inline task_DEV-1/FindMaxSequnceInString.cs]: useless comment

File: task_DEV-1/FindMaxSequnceInString.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
+﻿using Task1;
+
+namespace equal_sumbols
+{
+    class FindMaxSequnceInString
```
Comment (AlexanderSoroka) [inline task_DEV-1/FindMaxSequnceInString.cs]: All public class and method must have description

File: task_DEV-1/FindMaxSequnceInString.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
+﻿using Task1;
+
+namespace equal_sumbols
```
Comment (AlexanderSoroka) [inline task_DEV-1/FindMaxSequnceInString.cs]: codestyle violation

File: task_DEV-1/FindMaxSequnceInString.cs [hunk 11]
```diff
@@ -0,0 +1,40 @@
+﻿using Task1;
+namespace equal_sumbols
+{
+    class FindMaxSequnceInString //finds the maximum subseqence of characters in the string
```
Comment (AlexanderSoroka) [inline task_DEV-1/FindMaxSequnceInString.cs]: missed comment for public class

**Pull request merged**